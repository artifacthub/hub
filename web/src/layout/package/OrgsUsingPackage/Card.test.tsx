import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import OrgCard from './Card';

const defaultProps = {
  organization: {
    name: 'artifacthub',
    displayName: 'Artifact Hub',
    homeUrl: 'https://artifacthub.io',
    logoImageId: '35c5d3c7-421e-4b4a-97b0-86f4602a2f59',
  },
};

describe('OrgCard', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(
      <Router>
        <OrgCard {...defaultProps} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders component', () => {
      render(
        <Router>
          <OrgCard {...defaultProps} />
        </Router>
      );

      const org = screen.getByTestId('org-using-pkg');
      expect(org).toBeInTheDocument();
      expect(org).toHaveTextContent('Artifact Hub');

      const link = screen.getByRole('button', { name: 'Open organization url' });
      expect(link).toBeInTheDocument();
      expect(link).toHaveAttribute('href', 'https://artifacthub.io');
    });

    it('renders component without url', () => {
      render(
        <Router>
          <OrgCard organization={{ ...defaultProps.organization, homeUrl: undefined }} />
        </Router>
      );

      expect(screen.queryByRole('button', { name: 'Open organization url' })).toBeNull();
    });
  });
});
