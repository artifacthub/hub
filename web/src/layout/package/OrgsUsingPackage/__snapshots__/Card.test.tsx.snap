// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgCard creates snapshot 1`] = `
<DocumentFragment>
  <a
    aria-label="Open organization url"
    class="link d-inline-block text-decoration-none text-dark h5 mb-2 mr-3"
    href="https://artifacthub.io"
    rel="noopener noreferrer"
    role="button"
    tabindex="-1"
    target="_blank"
  >
    <div
      class="badge badge-rounded badge-light rounded-pill d-flex flex-row align-items-center pl-0 pr-3 badge"
      data-testid="org-using-pkg"
    >
      <div
        class="mr-2"
      >
        <div
          class="imageWrapper imageWrapper overflow-hidden"
        >
          <div
            class="d-flex align-items-center justify-content-center w-100 h-100"
          >
            <img
              alt="Artifact Hub"
              aria-hidden="true"
              class="m-auto image"
              src="/image/35c5d3c7-421e-4b4a-97b0-86f4602a2f59"
              srcset="/image/35c5d3c7-421e-4b4a-97b0-86f4602a2f59@1x 1x, /image/35c5d3c7-421e-4b4a-97b0-86f4602a2f59@2x 2x, /image/35c5d3c7-421e-4b4a-97b0-86f4602a2f59@3x 3x, /image/35c5d3c7-421e-4b4a-97b0-86f4602a2f59@4x 4x"
            />
          </div>
        </div>
      </div>
      <div
        class="text-truncate badgeContent"
      >
        <span>
          Artifact Hub
        </span>
      </div>
      <small
        class="ml-2"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <polyline
            points="15 3 21 3 21 9"
          />
          <line
            x1="10"
            x2="21"
            y1="14"
            y2="3"
          />
        </svg>
      </small>
    </div>
  </a>
</DocumentFragment>
`;
